packages/phpstan-rules/tests/Rules/PreventDoubleSetParameterRule/Fixture/DuplicateConstantValue.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symplify\PHPStanRules\Tests\Rules\PreventDoubleSetParameterRule\Source\OptionConstants;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $parameters = $containerConfigurator->parameters();
+    $parameters->set(OptionConstants::NAME, 'b');
+    $parameters->set(OptionConstants::NAME, 'c');
+};
```

Comment:
@samsonasik The rule does not work for constants. This test needs to pass.
Could you take over this PR and add this feature?

Partialy `constExprEvaluator` can be used, but some extra code is needed.

This is how Rector handles it:

https://github.com/rectorphp/rector/blob/f3decc4a9c45bd85a21f5a8ee985a1f07f120ce7/src/PhpParser/Node/Value/ValueResolver.php#L203-L235

Feel free to copy paste is to some service here :)

Reply:
I will try.

Reply:
I added ClassConstFetch check, I only check with. Getting `ValueResolver` dependencies services functionality seems needs many copy paste. I added check with the following:

```php
$node instanceof ClassConstFetch && $node->class instanceof FullyQualified && $node->name instanceof Identifier
```

and it seems working ok to get `FQCN\The\Class::CONSTANT` string.

Reply:
Thank you :+1: